docs/TPM_with_LPC_protocol.md

```diff
@@ -0,0 +1,820 @@
+        .SPIm_PWdata_11S           ( 1'b0                           ), // input
+        .SPIm_PWdata_0S            ( 1'b0                           ), // input
+        .SPIm_Paddr_8S             ( 1'b0                           ), // input
+        .SPIm_Paddr_6S             ( 1'b0                           ), // input
+        .FB_PKfbPush_1S            ( 1'b0                           ), // input
+        .FB_PKfbData_31S           ( 1'b0                           ), // input
+        .FB_PKfbData_21S           ( 1'b0                           ), // input
+        .FB_PKfbData_19S           ( 1'b0                           ), // input
+        .FB_PKfbData_9S            ( 1'b0                           ), // input
+        .FB_PKfbData_6S            ( 1'b0                           ), // input
+        .Sys_PKfb_ClkS             ( 1'b0                           ), // input
+        .FB_BusyS                  ( 1'b0                           ), // input
+        .WB_CLKS                   ( 1'b0                           )  // input
+
+    );
+```
+As one can see there is much more thant only clocks. There is one big disadvantage
+related to using "cell_macro" construct. This "cell-macro" is just a "black-box"
+and we haven't any model how it works. This fact makes impossible simulation of 
+FPGA part of application for SoC "EOS S3", so we can;t determine this way if 
+application is working properly.
+We just check if communication between FPGA and ARM Cortex-M4 MCU using "Wishbone Bus"
+to "AHB" bridge and interrupts using hardware.
+In Verilog source file "AL4S3B_FPGA_ONION_LPCCTRL.v" - see link:
+["EOS S3 SoC application module"](https://github.com/lpn-plant/lpntpm-lpc/blob/main/SOC_EOS_S3_Application_Test_comunication/fpga/rtl/AL4S3B_FPGA_ONION_LPCCTRL.v)
+we declared three 32-bit registers:
+```verilog
+reg     [31:0]  BREATHE_0_CONFIG   = 32'h00000010;         
+reg     [31:0]  BREATHE_1_CONFIG   = 32'h00002000;
+reg     [31:0]  BREATHE_2_CONFIG   = 32'h03000000;     
+```
+Then in always block:
+```verilog
+     //-----------------------------------------------
+       cnt3 = cnt3 + 1;
+       if ((cnt3 >= 1024000) && (cnt3 < 1024900))
+       begin //period 1.25s
+	     if (cnt3==1024000)
+	     begin 	 
+    	   BREATHE_0_CONFIG = BREATHE_0_CONFIG + 1;
+    	   BREATHE_1_CONFIG = BREATHE_1_CONFIG + 1;
+    	   BREATHE_2_CONFIG = BREATHE_2_CONFIG + 1;
+    	 end  
+    	 TIMER_o = 4'b1111;  //activate interrupt for MCU part
+       end
+       else if (cnt3 >= 1024900)
+       begin
+	      cnt3 = 20'h00000;
+	      TIMER_o = 4'b0000;  //deactivate interrupt for MCU part	   
+       end
+    end  
+```
+we periodically increment these register and set TIMER_o = 4'b1111; signal which
+is interrupt vector passed to MCU. When MCU get this inttrupt it programatically
+reads these registers by internal "Wishbone Bus" 2 "AHB" bridge. After short time
+the interrupt in FPGA is deactivated: TIMER_o = 4'b0000;
```

Comment:
@mgabryelski1 Why is the interrupt deactivated automatically? Shouldn't it be deactivated after host clears it?